src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Verbs.tt

```diff
@@ -242,8 +188,66 @@ foreach (XElement verb in commands.Descendants(XName.Get("Verb","")))
                 <#= commandFormalName #>Cmd.Description = Strings.<#= commandFormalName #>_Description;
                 <#= commandFormalName #>Cmd.OnExecute(() =>
                 {
+<#
+        bool verbHasOptions = false;
+        foreach (XElement option in verb.Descendants())
+        {
+            if (option.Name == XName.Get("Noun"))
+            {
+                break;
+            }
+
+            if (!verbHasOptions)
+            {
+#>
+                    var args = new <#= commandFormalName #>Args()
+                    {
+<#
+            }
+
+            verbHasOptions = true;
+            var optionData = new OptionData(option);
+            if (optionData.BaseClass == "Argument")
+            {
+#>
+                        <#= optionData.InitCapsName #> = <#= optionData.FormalName #>.Value,
+<#
+            }
+            else if (optionData.BaseClass == "Option")
+            {
+                switch (optionData.OptionType)
+                {
+                    case "SingleValue":
+#>
+                        <#= optionData.InitCapsName #> = <#= optionData.FormalName #>.Value(),
+<#
+                        break;
+                    case "NoValue":
+#>
+                        <#= optionData.InitCapsName #> = <#= optionData.FormalName #>.HasValue(),
+<#
+                        break;
+                }
+            }
+        }
+
+        if (verbHasOptions)
+        {
+#>
+                    };
+
+                    <#= commandFormalName #>Runner.Run(args, getLogger);
```

Comment:
I have noticed that [`VerifyCommandRunner` has instance method `ExecuteCommandAsync` ](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/VerifyCommand/VerifyCommandRunner.cs#L28) method instead of a `static.Run` method for e.g. [SourceRunners.cs](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs#L15)

Reply:
Do we end up with 2 patterns or just 1? (2 might be fine)
You could decide to support this with an attribute on a verb in command.xml, perhaps. Maybe something like Runner="async" or Runner="ExecuteCommandAsync" or similar.
Please do this as part of your work, and let's complete this PR if it meets its goal.